grow/pods/document_front_matter.py

```diff
@@ -79,7 +79,12 @@ def _load_front_matter(self, raw_front_matter):
                 self._doc.raw_content)
 
         if self._raw_front_matter:
-            _update_deep(self.data, self._load_yaml(self._raw_front_matter))
+            new_data = self._load_yaml(self._raw_front_matter)
+            if not isinstance(new_data, dict):
+                raise BadFormatError(
+                    'Front matter needs to be a dictionary, found <{}>'.format(
```

Comment:
Show the offending document's pod path in the error message too - or will it already be displayed when the error bubbles up?